### PR TITLE
fix(docs) Corrected challenge template example

### LINKED ISSUE #34208: Challenge template incorrect
**Describe your problem and how to reproduce it:**
I think the challenge template in the "How to work on coding challenges" guide [here](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-work-on-coding-challenges.md) is incorrect . I think there should be these lines:
```
## Challenge Seed
<section id='challengeSeed'>
```
Before this:
```
<div id='js-seed'>
```
Also there are closed section tags without corresponding open tags:
````
### Before Test
<div id='js-setup'>

```js
Test setup code.
 ```
</div>

</section>

### After Test
<div id='js-teardown'>

```js
Test tear down code.
```

</div>

</section>
````
**Add a Link to the page with the problem:**
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-work-on-coding-challenges.md

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

I fixed the issue below and also attempt to resolve the suggestions requested by @ValeraS on an earlier PR which which became stale.

Closes #34208

### CODE REVIEW

**Review by thecodingaviator — APPROVED** [review]
Looks good

File: docs/how-to-work-on-coding-challenges.md [hunk 1]
```diff
@@ -32,40 +36,46 @@ A Description of the challenge and what is required to pass
...
 Instructions about what exactly needs to be done.
 </section>
+
 ## Tests
 <section id='tests'>
 
 ``` yml
 - text: Should return "foo".
-  testString: 'A stringified function using Chai asserts'
+  testString: 'A stringified function possibly using Chai asserts'
```
[the remark this answers is not in the document]
Comment (ValeraS) [reply docs/how-to-work-on-coding-challenges.md]: We should mention somehow: `text` and `testString` strings should be valid YAML strings

[the remark this answers is not in the document]
Comment (ValeraS) [reply docs/how-to-work-on-coding-challenges.md]: A stringified function or expression possibly using Chai asserts

**Review by ValeraS — CHANGES_REQUESTED** [review]

**Review by ValeraS — CHANGES_REQUESTED** [review]
And check open/close tags

**Comment by RandellDawson** [comment]
@ValeraS OK, I think I covered everything you requested.  I put some more notes near the top regarding the `Tests` section.

File: docs/how-to-work-on-coding-challenges.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ValeraS) [reply docs/how-to-work-on-coding-challenges.md]: @RandellDawson I did not mean to replace text to `bla bla`. I have shown missing line `tests:` and indentations.

**Comment by ValeraS** [comment]
> And check open/close tags

Remove `</section>` before `### After tests

**Comment by RandellDawson** [comment]
@ValeraS So sorry about that.  I must have a looked at that 10 times and did not see it.

**Review by raisedadead — APPROVED** [review]

**Review by ValeraS — APPROVED** [review]

**Pull request merged**